src/SFA.DAS.ApprenticeCommitments.Web/Identity/RequiresIdentityConfirmed.cs

```diff
@@ -15,17 +15,19 @@ public RequiresIdentityConfirmedAttribute() : base(typeof(ClaimRequirementFilter
     public class ClaimRequirementFilter : IAuthorizationFilter
     {
         private readonly NavigationUrlHelper _urlHelper;
+        private readonly AuthenticatedUser _user;
 
-        public ClaimRequirementFilter(NavigationUrlHelper urlHelper)
+        public ClaimRequirementFilter(NavigationUrlHelper urlHelper, AuthenticatedUser user)
         {
             _urlHelper = urlHelper;
+            _user = user;
         }
         public void OnAuthorization(AuthorizationFilterContext context)
         {
-            if (context.HttpContext.UserMustCreateAccount())
+            if (!_user.HasCreatedAccount)
```

Comment:
I quite like having the predicates be positive - reading "if user must create account" seems more natural than "if not user has created account"